File: 5° semestre/Programação para dispositivos moveis II/aula-25-09-26/atividades pré prova/1 - ComprasCheckbox/projeto1/app/src/main/java/br/gov/sp/cps/comprascheckbox/MainActivity.java
package br.gov.sp.cps.comprascheckbox;

import android.os.Bundle;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private CheckBox cb1, cb2, cb3, cb4, cb5;
    private Button btnTotal;
    private TextView textTotal;
    private double[] prices = {2.69, 2.70, 16.70, 3.38, 3.00};
    private double total;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        cb1 = findViewById(R.id.cb1);
        cb2 = findViewById(R.id.cb2);
        cb3 = findViewById(R.id.cb3);
        cb4 = findViewById(R.id.cb4);
        cb5 = findViewById(R.id.cb5);
        btnTotal = findViewById(R.id.btnTotal);
        textTotal = findViewById(R.id.textTotal);

        btnTotal.setOnClickListener(v -> {
            total = 0.00;
            if(cb1.isChecked()){
            total += prices[0];
        }
            if(cb2.isChecked()){
                total += prices[1];
            }
            if(cb3.isChecked()){
                total += prices[2];
            }
            if(cb4.isChecked()){
                total += prices[3];
            }
            if(cb5.isChecked()){
                total += prices[4];
            }
            textTotal.setText(String.format("R$ %.2f", total));
        });
    }
}

File: 3° semestre/Programação para dispositivos Moveis I/aula-24-11-19/calculadoraBitcoin/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import java.net.URL
import android.widget.TextView
import kotlinx.coroutines.*
import org.json.JSONObject
import java.text.NumberFormat
import java.util.*

class MainActivity : AppCompatActivity() {

    val API_URL = "https://www.mercadobitcoin.net/api/BTC/ticker/"

    var cotacaoBitcoin: Double = 0.0

    // Variáveis de views
    private lateinit var btnCalcular: TextView
    private lateinit var txtCotacao: TextView
    private lateinit var txtValor: TextView
    private lateinit var txtQtdBitcoins: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Inicializando as views
        btnCalcular = findViewById(R.id.btnCalcular)
        txtCotacao = findViewById(R.id.txtCotacao)
        txtValor = findViewById(R.id.txtValor)
        txtQtdBitcoins = findViewById(R.id.txtQtdBitcoins)

        buscarCotacao()

        btnCalcular.setOnClickListener {
            calcular()
        }
    }

    fun buscarCotacao() {
        // Usando CoroutineScope para controle correto da thread
        CoroutineScope(Dispatchers.IO).launch {
            // Trabalha em background
            val resposta = URL(API_URL).readText()
            cotacaoBitcoin = JSONObject(resposta).getJSONObject("ticker")
                .getDouble("last")

            val f = NumberFormat.getCurrencyInstance(Locale("pt", "BR"))
            val cotacaoFormatada = f.format(cotacaoBitcoin)

            withContext(Dispatchers.Main) {
                // Atualizando a UI com a cotação formatada
                txtCotacao.text = cotacaoFormatada
            }
        }
    }

    fun calcular() {
        if (txtValor.text.isEmpty()) {
            txtValor.error = "Preencha um valor"
            return
        }
        // Valor digitado pelo usuário
        val valorDigitado = txtValor.text.toString()
            .replace(",", ".").toDouble()

        // Caso não venha valor da API
        val resultado = if (cotacaoBitcoin > 0) valorDigitado / cotacaoBitcoin else 0.0

        // Atualizando o TextView com o resultado formatado com 8 casas decimais
        txtQtdBitcoins.text = "%.8f".format(resultado)
    }
}
